examples/tv-casting-app/tv-casting-common/core/CastingPlayerDiscovery.h

```diff
@@ -0,0 +1,117 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include "CastingPlayer.h"
+#include "Types.h"
+
+#include <controller/CHIPCommissionableNodeController.h>
+#include <controller/DeviceDiscoveryDelegate.h>
+#include <vector>
+
+namespace matter {
+namespace casting {
+namespace core {
+
+/**
+ * @brief Represents CastingPlayerDiscovery state.
+ *
+ */
+enum CastingPlayerDiscoveryState
+{
+    DISCOVERY_NOT_READY, // Default state, mClientDelegate isn't initialized
+    DISCOVERY_READY,     // After SetDelegate and before StartDiscovery, mClientDelegate is initialized
+    DISCOVERY_RUNNING,   // After StartDiscovery success
+};
+
+class DLL_EXPORT DiscoveryDelegate
+{
+public:
+    virtual ~DiscoveryDelegate() {}
+    virtual void HandleOnAdded(Strong<CastingPlayer> player)    = 0;
+    virtual void HandleOnUpdated(Strong<CastingPlayer> players) = 0;
+    // virtual void HandleOnRemoved(std::vector<Strong<CastingPlayer>> players);
+};
+
+class CastingPlayerDiscovery;
+
+class DeviceDiscoveryDelegateImpl : public chip::Controller::DeviceDiscoveryDelegate
+{
+private:
+    DiscoveryDelegate * mClientDelegate = nullptr;
+
+public:
+    DeviceDiscoveryDelegateImpl() {}
+    DeviceDiscoveryDelegateImpl(DiscoveryDelegate * delegate) { mClientDelegate = delegate; }
+
+    void OnDiscoveredDevice(const chip::Dnssd::DiscoveredNodeData & nodeData) override;
+};
+
+class CastingPlayerDiscovery
```

Comment:
nit: Can we move this class up to the top of this file (and forward declare any of the classes it needs) since this is the main one for it ?

Reply:
done